central/externalbackups/plugins/s3/s3.go

```diff
@@ -121,29 +152,34 @@ func sortS3Objects(objects []*awsS3.Object) {
 }
 
 func (s *s3) pruneBackupsIfNecessary() error {
```

Comment:
You can pass a context here to avoid `context.Background` (see `s3compatible` implementation).

Reply:
I'll do another pass on that PR to make it closer to `s3compatible`.
The extraction of common code will be done in #15300